### PR TITLE
[EC-154] [BEEEP] Remove factory providers in Angular DI

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

See https://github.com/bitwarden/jslib/pull/775.

## Code changes

* update `services.module.ts` to use the new tokens
* extract init logic to an `initService` so that it's not mixed in with provider logic
* add `@Injectable()` to `WebPlatformUtilsService` so that we don't have to specify its deps. I was a little unsure about mixing patterns/approaches, but I settled on making use of Angular's features where we can.

## Before you submit

- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: src/app/services/services.module.ts
Comment (Hinton) [inline src/app/services/services.module.ts:76]: Thoughts on moving `"locales"` to a variable?

Comment (eliykat) [reply src/app/services/services.module.ts]: I assume you mean token - done, depends on https://github.com/bitwarden/jslib/pull/783

**Review by Hinton — COMMENTED** [review]
LGTM once we merge the jslib pr.

**Review by Hinton — APPROVED** [review]

**Pull request merged**
